Chocolatey updates broken

Looks like Chocolatey updates stopped updating at `v0.2.9` - https://chocolatey.org/packages/act-cli#versionhistory.

I'm going to see if there is anything obvious that can be addressed there.

Looking at the error logs for the most recent push - the [first error I see](https://github.com/nektos/act/runs/1370743179#step:4:551) is:

```
              run_tasks:
              
                   [echo] ====================
                   [echo] GitVersioner
                   [echo] ====================
              Running custom tasks if /usr/local/src/choco/.build.custom/versioners/git.pre.step exists (or with .ps1/.rb added to the end of it).
              Setting app.ruby to 'ruby'
              Setting app.powershell to '/System32/WindowsPowerShell/v1.0/powershell.exe'
              Running custom tasks if /usr/local/src/choco/.build.custom/versioners/git.replace.step exists (or with .ps1/.rb added to the end of it).
              Setting app.ruby to 'ruby'
              Setting app.powershell to '/System32/WindowsPowerShell/v1.0/powershell.exe'
              
              get_revision:
              
                   [echo] Getting revision number from working directory '/usr/local/src/choco' or remotely at https://github.com/chocolatey/choco.
                   [exec] Starting 'git (describe)' in '/usr/local/src/choco'
                   [exec] /usr/local/src/choco/.build/versioners/git.step(26,6):
                   [exec] 'git' failed to start.
              
              BUILD FAILED - 1 non-fatal error(s), 4 warning(s)
              
              /usr/local/src/choco/.build/versioners/git.step(40,7):
              Property evaluation failed.
              Expression: ${string::trim(version.revision.output)}
                                         ^^^^^^^^^^^^^^^^^^^^^^^:
              NAnt.Core.BuildException: /usr/local/src/choco/.build/versioners/git.step(40,7):
              Property evaluation failed.
              Expression: ${string::trim(version.revision.output)}
                                         ^^^^^^^^^^^^^^^^^^^^^^^ ---> Property 'version.revision.output' has not been set.:
              NAnt.Core.BuildException: Property 'version.revision.output' has not been set.
                at NAnt.Core.ExpressionEvaluator.GetPropertyValue (System.String propertyName) [0x00000] in <filename unknown>:0 
                at NAnt.Core.ExpressionEvaluator.EvaluateProperty (System.String propertyName) [0x00000] in <filename unknown>:0 
                at NAnt.Core.ExpressionEvalBase.ParseValue () [0x00000] in <filename unknown>:0 
                --- End of inner exception stack trace ---
                at NAnt.Core.PropertyDictionary.EvaluateEmbeddedExpressions (System.String input, NAnt.Core.Location location, System.Collections.Hashtable state, System.Collections.Stack visiting) [0x00000] in <filename unknown>:0 
                at NAnt.Core.PropertyDictionary.ExpandProperties (System.String input, NAnt.Core.Location location, System.Collections.Hashtable state, System.Collections.Stack visiting) [0x00000] in <filename unknown>:0 
                at NAnt.Core.PropertyDictionary.ExpandProperties (System.String input, NAnt.Core.Location location) [0x00000] in <filename unknown>:0 
                at NAnt.Core.Project.ExpandProperties (System.String input, NAnt.Core.Location location) [0x00000] in <filename unknown>:0 
                at NAnt.Core.Tasks.PropertyTask.ExecuteTask () [0x00000] in <filename unknown>:0 
                at NAnt.Core.Task.Execute () [0x00000] in <filename unknown>:0 
              
              Total time: 1.2 seconds.
              
       [nant] Nested build failed.  Refer to build log for exact reason.:
       [nant] NAnt.Core.BuildException: Nested build failed.  Refer to build log for exact reason.
       [nant]   at NAnt.Core.Tasks.NAntTask.RunBuild (System.IO.FileInfo buildFile) [0x00000] in <filename unknown>:0 
       [nant]   at NAnt.Core.Tasks.NAntTask.ExecuteTask () [0x00000] in <filename unknown>:0 
       [nant]   at NAnt.Core.Task.Execute () [0x00000] in <filename unknown>:0 
       [echo] Using revision number: 0 and hash: 0.
```

Adding `git` to the apt-get install of the Dockerfile resolves the error above but produces a new error because the working directory isn't a git repository:

```get_revision:
            
                 [echo] Getting revision number from working directory '/usr/local/src/choco' or remotely at https://github.com/chocolatey/choco.
                 [exec] Starting 'git (describe)' in '/usr/local/src/choco'
                 [exec] fatal: Not a git repository (or any of the parent directories): .git
                 [exec] /usr/local/src/choco/.build/versioners/git.step(26,6):
                 [exec] External Program Failed: git (return code was 128)
               [delete] Deleting file /usr/local/src/choco/.build/versioners/_revision.xml.
            
            get_externally_provided_revision:
            
            
            set_version_environment_variable:
            
               [setenv] Setting environment variable "uc.app.revision" to "fatal: Not a git repository (or any of the parent directories): .git".
               [setenv] Setting environment variable "uc.app.revision.hash" to "fatal: Not a git repository (or any of the parent directories): .git".
            
            run_normal_tasks:
            
            Running custom tasks if /usr/local/src/choco/.build.custom/versioners/git.post.step exists (or with .ps1/.rb added to the end of it).
            Setting app.ruby to 'ruby'
            Setting app.powershell to '/System32/WindowsPowerShell/v1.0/powershell.exe'
            
            go:
            
            
            BUILD SUCCEEDED - 1 non-fatal error(s), 5 warning(s)
            
            Total time: 0.9 seconds.
```

I'm not sure if this is the error that is ultimately leading to the build failing however.  Here are some examples of the errors that ultimately cause the build to fail I believe:

```                 [exec]  infrastructure/registration/SimpleInjectorContainer.cs(32,38): error CS0246: The type or namespace name `Container' could not be found. Are you missing `System.ComponentModel' using directive?
                 [exec]  infrastructure/registration/SimpleInjectorContainer.cs(62,23): error CS0246: The type or namespace name `Container' could not be found. Are you missing `System.ComponentModel' using directive?
                 [exec]  infrastructure/registration/SimpleInjectorContainer.cs(67,24): error CS0246: The type or namespace name `Container' could not be found. Are you missing `System.ComponentModel' using directive?
                 [exec]  infrastructure/registration/SimpleInjectorContainer.cs(92,77): error CS0246: The type or namespace name `Container' could not be found. Are you missing `System.ComponentModel' using directive?
                 [exec]  infrastructure/logging/Log4NetAppenderConfiguration.cs(39,33): error CS0246: The type or namespace name `log4net' could not be found. Are you missing an assembly reference?
                 [exec]  infrastructure/logging/Log4NetLog.cs(35,17): error CS0246: The type or namespace name `log4net' could not be found. Are you missing an assembly reference?
                 [exec]  infrastructure/logging/Log4NetLog.cs(115,26): error CS0246: The type or namespace name `Level' could not be found. Are you missing an assembly reference?```

--- a/.github/actions/choco/Dockerfile
+++ b/.github/actions/choco/Dockerfile
@@ -1,8 +1,12 @@
-  
-FROM mono:3.12.1 as builder
-ARG CHOCOVERSION=0.10.15
+# choco supports only Mono 5.20.x
+# https://github.com/chocolatey/choco#prerequisites
+FROM mono:5.20 as builder
+# 0.10.15 with newer Mono is failing as it's dependent on Mono Profile 4.0 which has been deprecated
+# http://www.mono-project.com/docs/about-mono/releases/4.0.0/#dropped-support-for-old-frameworks
+# Alternative to consider, use git repository and reset to specific tag/hash
+ARG CHOCOVERSION=0.10.16-beta
 
-RUN echo "deb http://archive.debian.org/debian/ wheezy main contrib non-free" >/etc/apt/sources.list
+ENV DEBIAN_FRONTEND=noninteractive
 RUN apt-get update && apt-get install -y wget tar gzip
 
 WORKDIR /usr/local/src
@@ -30,5 +34,3 @@ RUN apk --update --no-cache --repository http://dl-cdn.alpinelinux.org/alpine/ed
 
 COPY entrypoint.sh /entrypoint.sh
 ENTRYPOINT ["/entrypoint.sh"]
-
-

